src/status_im/ui/screens/chat/message/message.cljs

```diff
@@ -298,28 +298,45 @@
                                          {:content (sheets/sticker-long-press message)
                                           :height  64}])}))
 
-(defn chat-message [{:keys [public? content content-type] :as message}]
-  (if (= content-type constants/content-type-command)
-    [message.command/command-content message-content-wrapper message]
-    (if (= content-type constants/content-type-system-text)
-      [system-message-content-wrapper message [system-text-message message]]
-      [message-content-wrapper
-       message
-       (if (= content-type constants/content-type-text)
-         ;; text message
-         [text-message message]
-         (if (= content-type constants/content-type-status)
-           [message-content-status message]
-           (if (= content-type constants/content-type-emoji)
-             [emoji-message message]
-             (if (= content-type constants/content-type-sticker)
-               [react/touchable-highlight (sticker-message-press-handlers message)
-                [react/image {:style  {:margin 10 :width 140 :height 140}
-                              ;;TODO (perf) move to event
-                              :source {:uri (contenthash/url (-> content :sticker :hash))}}]]
-               (if (and (= content-type constants/content-type-image)
-                        ;; Disabling images for public-chats
-                        (not public?))
-                 [react/touchable-highlight (image-message-press-handlers message)
-                  [message-content-image message]]
-                 [unknown-content-type message])))))])))
+(defmulti ->message :content-type)
```

Comment:
this one is really critical from performance, so here we should write code for better performance not readability

Reply:
I will benchmark it

Reply:
Percentile to milliseconds, profiled using https://reactjs.org/docs/profiler.html and https://github.com/welch/tdigest

Multimethod in this branch(removed the reactions modal to lower the code difference):
```
[Sun Jul 05 2020 17:22:19.205]  INFO     INFO [status-im.utils.profiler:23] - [profile: Messsage ( mount )]: 
 approximating 685 samples using 52 centroids
0 = 0
0.25  = 7.368131868131869
0.5  = 9.883040935672515
0.75  = 13.682692307692308
1.0 = 241
```

Develop:
```
[Sun Jul 05 2020 17:24:32.855]  INFO     INFO [status-im.utils.profiler:23] - [profile: Messsage ( mount )]: 
 approximating 654 samples using 46 centroids
0 = 0
0.25  = 7.326086956521739
0.5  = 10.205882352941176
0.75  = 14.872727272727273
1.0 = 82
```

The set of samples messages is quite small but we can already see that this branch has quite better performance then develop.

Reply:
not sure about comparing brunch but all benchmark says multimethods are slow https://github.com/alexkehayias/chocolatier#performance
`Don't use multimethod, use cond or condp`

Reply:
https://insideclojure.org/2015/04/27/poly-perf/
```

Expression | 1.6.0 | 1.7.0-beta2
-- | -- | --
(value-case 1) | 21 | 17
(value-case 5) | 26 | 18
(value-cond 1) | 4 | 7
(value-cond 5) | 90 | 74
(value-multi 1) | 41 | 44
(value-multi 5) | 47 | 40


```

Reply:
The worst-case with multimethod in JVM benchmark is 47ns. We have 16ms to render which is equal to 16000000. 16000000/47 so we can safely dispatch 340425 messages at once if we assume that it is the slowest part of the message render. We could definitely go with a case here which is transpiled into a switch, and we could gain constant time dispatching. But performance problems do not come from this part of code.